.github/workflows/build-docker.yaml

```diff
@@ -0,0 +1,182 @@
+---
+# The goal of this workflow is to build the `hoprd` docker image when a pull request is created or commit push into it and has the label `deploy_nodes`
+name: Build docker images
+
+on:
+  pull_request:
+    branches:
+      - master
+      - release/*
+    types:
+      - synchronize
+      - opened
+
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  change_set:
+    name: Get PR changeSet
+    runs-on: ubuntu-latest
+    outputs:
+      toolchain: ${{ steps.changeset.outputs.toolchain}}
+      hopli: ${{ steps.changeset.outputs.hopli}}
+      hoprd: ${{ steps.changeset.outputs.hoprd}}
+    steps:
+      - name: Checkout hoprnet
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get changeSet
+        id: changeset
+        # Set outputs using the command.
+        run: |
+          echo "toolchain=$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.ref }} ${{ github.event.pull_request.head.ref }} | grep -e ^scripts/ -e ^Makefile$ -e ^package.json$ -e ^.yarnrc.yml$ -e ^rust-toolchain.toml$ -e ^.nvmrc -e ^yarn.lock$ -e ^Cargo.toml$ | xargs)" >> $GITHUB_OUTPUT
+          echo "hopli=$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.ref }} ${{ github.event.pull_request.head.ref }} | grep ^packages/hopli/ | xargs)" >> $GITHUB_OUTPUT
+          echo "hoprd=$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.ref }} ${{ github.event.pull_request.head.ref }} | grep -v ^.github/ | grep ^packages/hopli/ | grep -v ^scripts | grep -v ^.processes | grep -v ^docs/ | grep -v .md  | xargs)" >> $GITHUB_OUTPUT
+
+  build_toolchain_image:
```

Comment:
Both images are similar, could it be possible to use the matrix?

Reply:
It might look similar but eventually they will have differences. In fact I haven't setup the same cache that @tolbrino  added, and it's something pending to do